### PR TITLE
Changed so saving if, no projectId is given, will create a new projec…

### PR DESCRIPTION
**Pull request by Daresoul**
[Syn-119](https://synthchron.atlassian.net/jira/software/projects/SYN/boards/1?selectedIssue=SYN-119)

## Describe your changes
Changed so it will now save and change the projectId, when saving a project without a projectId (either by being in the editor without a project loaded or joining a collaboration)

## Notable changes
Have changed `addProject` return type to be the projectId so the editor knows which project is currently being edited. It needs to know, so it wont create many different saves each time, but once its saved they can keep on saving onto it.

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

File: packages/synthchron-tauri/src/components/common/persistentStore.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ interface PersistentState {
   projects: { [id: string]: Project }
   saving: boolean
   doneSaving: () => void
-  addProject: (project: Project) => void
+  addProject: (project: Project) => number
```
Comment (leny-mi) [inline packages/synthchron-tauri/src/components/common/persistentStore.ts:9]: A project id is a string, not a number

**Review by leny-mi — APPROVED** [review]
looks good, but we need to change the addProject signature to return a string instead of a number, as this might otherwise cause issues in the future

File: packages/synthchron-tauri/src/components/common/persistentStore.ts (hunk 1 above)
Comment (Daresoul) [reply packages/synthchron-tauri/src/components/common/persistentStore.ts:9]: Gonna change this

**Review by leny-mi — APPROVED** [review]
lgtm

**Pull request merged**